repository/src/main/java/org/apache/atlas/accesscontrol/persona/AtlasPersonaService.java

```diff
@@ -0,0 +1,599 @@
+import static org.apache.atlas.accesscontrol.AccessControlUtil.getQualifiedName;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.getTenantId;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.getUUID;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.isDataPolicy;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.submitCallablesAndWaitToFinish;
+import static org.apache.atlas.accesscontrol.AccessControlUtil.validateUniquenessByName;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.ATTR_RANGER_ROLE_ID;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.CLASSIFICATION_ACTIONS;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.ENTITY_ACTIONS;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.LABEL_ACTIONS;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.LABEL_PREFIX_PERSONA;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.LABEL_TYPE_PERSONA;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.TERM_ACTIONS;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getLabelsForPersonaPolicy;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getPersonaLabel;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getPersonaPolicyLabel;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getPersonaRoleId;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getRoleName;
+import static org.apache.atlas.repository.Constants.PERSONA_ENTITY_TYPE;
+import static org.apache.atlas.repository.Constants.POLICY_ENTITY_TYPE;
+import static org.apache.atlas.repository.Constants.QUALIFIED_NAME;
+
+
+public class AtlasPersonaService {
+    private static final Logger LOG = LoggerFactory.getLogger(AtlasPersonaService.class);
+
+    private final AtlasGraph graph;
+    private final ESAliasStore aliasStore;
+    private final EntityGraphRetriever entityRetriever;
+
+    private static AtlasRangerService atlasRangerService = null;
+
+    public AtlasPersonaService(AtlasGraph graph, EntityGraphRetriever entityRetriever) {
+
+        this.graph = graph;
+        this.entityRetriever = entityRetriever;
+        this.aliasStore = new ESAliasStore(graph, entityRetriever);
+
+        atlasRangerService = new AtlasRangerService();
+    }
+
+    public void createPersona(PersonaContext context) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("createPersona");
+        LOG.info("Creating Persona");
+        AtlasEntityWithExtInfo entityWithExtInfo = context.getPersonaExtInfo();
+        context.setCreateNewPersona(true);
+
+        try {
+            validateUniquenessByName(graph, getName(entityWithExtInfo.getEntity()), PERSONA_ENTITY_TYPE);
```

Comment:
Why do we need this? won't this be automatically handled by qualifiedName + typeName

Reply:
this is uniqueness by name & not qualifiedName, so we have to handle it with our logic